sycl/include/CL/sycl/INTEL/esimd/detail/emu/esimdcpu_device_interface.hpp

```diff
@@ -0,0 +1,93 @@
+//==----- esimdcpu_device_interface.hpp - DPC++ Explicit SIMD API ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+/// \file esimdcpu_device_interface.hpp
+/// Declarations for ESIMD_CPU-device specific definitions.
+/// ESIMD intrinsic and LibCM functionalities required by intrinsic defined
+///
+/// This interface is for ESIMD intrinsic emulation implementations
+/// such as slm_access to access ESIMD_CPU specific-support therefore
+/// it has to be defined and shared as include directory
+///
+/// \ingroup sycl_pi_esimd_cpu
+
+#pragma once
+
+#include <CL/sycl/detail/plugin.hpp>
+#include <cstdint>
+
+/// This is the device interface version required (and used) by this implementation of
+/// the ESIMD CPU emulator.
+#define ESIMD_DEVICE_INTERFACE_VERSION 0
+
+#ifdef _MSC_VER
+// Definitions for type consistency between ESIMD_CPU and CM_EMU
+typedef unsigned int uint;
+typedef unsigned short ushort;
+typedef unsigned char uchar;
+#endif // _MSC_VER
+
+struct ESIMDDeviceInterface {
+  void *reserved;
+
+  ESIMDDeviceInterface();
+#include "esimd_emu_functions_v0.h"
+};
+
+
+// Denotes the data version used by the implementation.
+// Increment whenever the 'data' field interpretation within PluginOpaqueData is changed.
+#define ESIMD_EMU_PLUGIN_OPAQUE_DATA_VERSION 0
+
+// The table below shows the correspondence between the \c version
+// and the contents of the \c data field for ESIMD CPU Emulator plugin
+// instantiation of the \c PluginOpaqueData structure:
+// version == 0, data is ESIMDDeviceInterface*
+
+ESIMDDeviceInterface *getESIMDDeviceInterface() {
```

Comment:
I'm not if it's possible. As its name stands, the function is dedicated for ESIMD_CPU and calls PI_API for accessing opaque data from ESIMD_CPU library file.

Reply:
Not sure I understand how this blocks us from moving part of definition to the library (libsycl.so). There is anyway calls to this library from this function - cl::sycl::detail::pi::getPlugin<cl::sycl::backend::esimd_cpu>();

Reply:
Okay, I'll update this PR first without a change set for relocating this function. Then I'll post another update only for the relocation. Where do you think definition for the function should be? (e.g.: `pi.hpp`)?

Reply:
Hm, suggest placing to `pi.cpp` since it is one function only. If we end up having more ESIMD_CPU specific function we can create a separate helper file in `source` dir.

Reply:
I see. I'll apply the change later as I described above.